src/ray/raylet/agent_manager.cc

```diff
@@ -97,236 +60,55 @@ void AgentManager::StartAgent() {
-       runtime_env_config,
-       serialized_allocated_resource_instances,
-       job_id,
-       callback = std::move(callback)](const Status &status,
-                                       const rpc::GetOrCreateRuntimeEnvReply &reply) {
-        if (status.ok()) {
-          if (reply.status() == rpc::AGENT_RPC_STATUS_OK) {
-            callback(true,
-                     reply.serialized_runtime_env_context(),
-                     /*setup_error_message*/ "");
-          } else {
-            RAY_LOG(INFO) << "Failed to create runtime env for job " << job_id
-                          << ", error message: " << reply.error_message();
-            RAY_LOG(DEBUG) << "Serialized runtime env for job " << job_id << ": "
-                           << serialized_runtime_env;
-            callback(false,
-                     reply.serialized_runtime_env_context(),
-                     /*setup_error_message*/ reply.error_message());
-          }
-
-        } else {
-          RAY_LOG(INFO)
-              << "Failed to create runtime env for job " << job_id
-              << ", status = " << status
-              << ", maybe there are some network problems, will fail the request.";
-          RAY_LOG(DEBUG) << "Serialized runtime env for job " << job_id << ": "
-                         << serialized_runtime_env;
-          callback(false, "", "Failed to request agent.");
-        }
-      });
+  });
 }
 
-void AgentManager::DeleteRuntimeEnvIfPossible(
-    const std::string &serialized_runtime_env,
-    DeleteRuntimeEnvIfPossibleCallback callback) {
-  if (disable_agent_client_) {
-    RAY_LOG(ERROR)
-        << "Failed to delete runtime environment URI because the Ray agent couldn't be "
-           "started due to the port conflict. See `dashboard_agent.log` for more "
-           "details. To solve the problem, start Ray with a hard-coded agent port. `ray "
-           "start --dashboard-agent-grpc-port [port]` and make sure the port is not used "
-           "by other processes.";
-    RAY_LOG(DEBUG) << "Serialized runtime env for failed URI deletion: "
-                   << serialized_runtime_env;
-    delay_executor_([callback = std::move(callback)] { callback(false); }, 0);
-    return;
-  }
-  // `runtime_env_agent_client_` should be `nullptr` when the agent is starting or the
-  // agent has failed.
-  if (runtime_env_agent_client_ == nullptr) {
-    RAY_LOG(INFO) << "Runtime env agent is not registered yet. Will retry "
-                     "DeleteRuntimeEnvIfPossible later.";
-    delay_executor_(
-
-        [this, serialized_runtime_env, callback = std::move(callback)] {
-          DeleteRuntimeEnvIfPossible(serialized_runtime_env, callback);
-        },
-        RayConfig::instance().agent_manager_retry_interval_ms());
-    return;
+AgentManager::~AgentManager() {
+  RAY_LOG(INFO) << "Killing agent " << options_.agent_name << ", pid " << process_.GetId()
+                << ".";
+  kill_raylet_if_process_exits_ = false;
```

Comment:
is this line necessary (since we anyway join the monitor thread?)?

Reply:
we don't want to let the monitor thread kill raylet because of raylet killing the agent.